org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaJdiEventDispatcher.scala

```diff
@@ -156,8 +166,15 @@ private class ScalaJdiEventDispatcherActor private (scalaDebugTargetActor: Suppr
         case result: Boolean => staySuspended |= result
       }
     }.andThen {
-      try if (!staySuspended) eventSet.resume()
-      finally ScalaJdiEventDispatcherActor.this ! FutureComputed
+      val resume = !staySuspended
+      val replyTo = this
+      Future {
+        jdiSynchronized {
```

Comment:
I'm not sure this is actually necessary, but I guess the aspect is very pick and expects every call in `com.sun.jdi` to be synched...